src/main/java/ch/ethz/seb/sebserver/webservice/servicelayer/lms/impl/olat/OlatLmsAPITemplate.java

```diff
@@ -332,18 +428,18 @@ private OAuth2RestTemplate createRestTemplate(final String accessTokenRequestPat
                 .getOrThrow();
 
         final ClientCredentialsResourceDetails details = new ClientCredentialsResourceDetails();
-        details.setAccessTokenUri(lmsSetup.lmsApiUrl + accessTokenRequestPath);
+        details.setAccessTokenUri(lmsSetup.lmsApiUrl + "/restapi/auth/");
         details.setClientId(plainClientId.toString());
         details.setClientSecret(plainClientSecret.toString());
 
         final ClientHttpRequestFactory clientHttpRequestFactory = this.clientHttpRequestFactoryService
                 .getClientHttpRequestFactory(proxyData)
                 .getOrThrow();
 
-        final OAuth2RestTemplate template = new OAuth2RestTemplate(details);
+        final OlatLmsRestTemplate template = new OlatLmsRestTemplate(details);
         template.setRequestFactory(clientHttpRequestFactory);
 
-        return template;
+        return Result.of(template);

```

Comment:
Instead of using Result.of(template) here you can put the whole code within 

`
return Result.tryCatch(() -> {

    ... code goes here

    return template
}
`

This ensures that a potential runtime error is catch within and is returned with the Results error field for further error handling.